internal/provider/provider.go

```diff
@@ -108,9 +113,27 @@ func (p *tidbcloudProvider) Configure(ctx context.Context, req provider.Configur
 		return
 	}
 
+	// Create a new dedicated client and set it to the provider dedicated client
+	var dedicatedEndpoint = tidbcloud.DefaultDedicatedEndpoint
+	if os.Getenv(TiDBCloudDedicatedEndpoint) != "" {
+		dedicatedEndpoint = os.Getenv(TiDBCloudDedicatedEndpoint)
+	}
+	var iamEndpoint = tidbcloud.DefaultIAMEndpoint
+	if os.Getenv(TiDBCloudIAMEndpoint) != "" {
+		iamEndpoint = os.Getenv(TiDBCloudIAMEndpoint)
+	}
+	dc, err := NewDedicatedClient(publicKey, privateKey, dedicatedEndpoint, iamEndpoint, fmt.Sprintf("%s/%s", UserAgent, p.version))
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Unable to create client",
+			"Unable to create tidb dedicated client:\n\n"+err.Error(),
+		)
+		return
+	}
```

Comment:
1. I think we don't need to use the default endpoint here, just throw env value in and hide the logic inside the client. 
2. Why dedicated client need IAM endpoint? I think there should be a IAM client.
3. tidb dedicated -> TiDB Cloud Dedicated; plz make sure messages user can see is in formal format.